crates/interledger-packet/src/packet.rs

```diff
@@ -741,6 +757,33 @@ mod test_prepare {
         assert!(Prepare::try_from(prep).is_err());
     }
 
+    #[test]
+    fn test_invalid_ts() {
+        use std::convert::TryInto;
+        for i in 12..(12 + 17) {
+            let mut prep = BytesMut::from(PREPARE_BYTES);
+            prep[i] = 9; // convert a byte from the address to a junk character
+            let x = match Prepare::try_from(prep) {
+                Ok(x) => x,
+                Err(_) => {
+                    continue;
+                }
+            };
+
+            let built = PrepareBuilder {
+                amount: x.amount(),
+                expires_at: x.expires_at(),
+                execution_condition: x.execution_condition().try_into().unwrap(),
+                destination: x.destination(),
+                data: x.data(),
+            }
+            .build();
+
+            assert_eq!(x, built);
+            assert_eq!(BytesMut::from(x), BytesMut::from(built));
```

Comment:
I think this part could now be removed, replaced with:

```suggestion
            Prepare::try_from(prep).unwrap_err();
```

I guess for the future refactoring in #708 it might be useful to assert what does the error render as:

```suggestion
            let err = Prepare::try_from(prep).unwrap_err();
            assert_eq!("Some prefix: DateTime must be numeric", &err.to_string());
```